File: JavaScript/7js 1/filterLucky.js
/*
|--------------------------------------------------------------------------
| Find the lucky numbers
|--------------------------------------------------------------------------
|
*/

function filterLucky(x) {
   res = []
   let num = x.map(i => String(i))
   num.forEach(i => {
      if (i.search('7') >= 0) res.push(Number(i));
   });
   return res;
}


function filterLucky(x) {
   return x.filter(n => /7/.test(n))
}


function filterLucky(x) {
   return x.filter(x => x.toString().includes('7'));
}


function filterLucky(x) {
   return x.filter(el => `${el}`.indexOf(7) > -1);
}


console.log(filterLucky([1, 2, 3, 4, 5, 6, 7, 68, 69, 70, 15, 17]), [7, 70, 17]);
console.log(filterLucky([71, 9907, 69]), [71, 9907]);

File: Python/7py 2/filter_lucky.py
"""
|--------------------------------------------------------------------------
| Find the lucky numbers
|--------------------------------------------------------------------------
|
"""

def filter_lucky(lst):
   return list(filter(lambda i: '7' in str(i), lst))


def filter_lucky(lst):
   return [i for i in lst if '7' in str(i)]


def filter_lucky(lst):
   return [i for i in lst if str(i).find('7') != -1]


def filter_lucky(lst):
   return [i for i in lst if str(i).count('7') != 0]


print(filter_lucky([1, 2, 3, 4, 5, 6, 7, 68, 69, 70, 15, 17]), [7, 70, 17])
print(filter_lucky([71, 9907, 69]), [71, 9907])